pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -301,6 +301,103 @@ type MonitoringSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +optional
 	ExternalMgrPrometheusPort uint16 `json:"externalMgrPrometheusPort,omitempty"`
+	// CustomizedPrometheus points to a customized Ceph prometheus alerts
+	// +optional
+	// +nullable
+	CustomizedPrometheus *PrometheusRuleCustomized `json:"prometheusRule,omitempty"`
+}
+
+//PrometheusRuleCustomized keep customized alerts
+type PrometheusRuleCustomized struct {
+	// +optional
+	Alerts Alerts `json:"alerts,omitempty"`
+}
+
+//Alerts All prometheus alerts
+type Alerts struct {
+	// +optional
+	CephMgrIsAbsent CephNamespacedAlert `json:"cephMgrIsAbsent,omitempty"`
```

Comment:
Is there a way to disable a rule? Or you would have to change the rule settings so they effectively don't trigger, such as a high limit?

Reply:
Yes, I can think of a way to do that:

We can add another field to CephAlert struct called "Enabled" that his default value will be true.
If I'll take [CephMgrIsAbsent](https://github.com/rook/rook/blob/705d98fd9d042864920072c86c01383a64ca6c7c/pkg/operator/ceph/cluster/mgr/template/prometheusrule.yaml#L39-L49) template alert.
So, by adding the if statement: `{{- if .CephMgrIsAbsent.Enabled }}`, we can disable the rule (if the user will set CephMgrIsAbsent.Enabled to false):

```yaml
[...]
{{- if .CephMgrIsAbsent.Enabled }}
    - alert: CephMgrIsAbsent
      annotations:
        description: Ceph Manager has disappeared from Prometheus target discovery.
        message: Storage metrics collector service not available anymore.
        severity_level: {{ .CephMgrIsAbsent.SeverityLevel }}
        storage_type: ceph
      expr: |
        label_replace((up{job="rook-ceph-mgr"} == 0 or absent(up{job="rook-ceph-mgr"})), "namespace", "{{ .CephMgrIsAbsent.Namespace }}", "", "")
      for: {{ .CephMgrIsAbsent.For }}
      labels:
        severity: {{ .CephMgrIsAbsent.Severity }}
{{- end}}
[...]
```

Would like that I'll add this functionality also?

Reply:
That would be useful, but could certainly be a separate PR. It's generally also preferred for a boolean to default to `false` so that if it's not set it will naturally be interpreted as false. So I would suggest calling such a setting as `disabled` to be false by default.

Reply:
done